### ISSUE TITLE
Integration tests are not working locally (MacOS 10.14.6)

### ISSUE THREAD

**Issue by hamboomger** [HabitRPG/habitica#12733]
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### General Info
  * UUID: 80b61b73-2278-4947-b713-a10112cfe7f5
  * Browser: Irrelevant to the issue
  * OS: MacOS Mojave 10.14.6

### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)
It seems like running integration tests locally **without** a Docker environment results in a failure. Below are the steps that I took on a fresh Habitica project clone after a full factory reset of my Macbook Pro laptop.

0. Run `cp config.json.example config.json`
1. Run `npm install` to ensure some required libs like `run-rs` are in place.
2. [Optional] Run `docker-compose -f docker-compose.dev.yml up -d` to see if the app works locally, followed by `docker-compose stop` after ensuring everything is OK.
3. Run `killall node`, as per the step-by-step guide [here](https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API) 
4. Run `npm run mongo:dev`, which results in an error
5. Run `npm run test:api-v3:integration`, which also results in an error because of the mongo db.

I also did not install the mongodb service on my laptop.

#### Console Errors
[//]: # (Include any JavaScript console errors here.)
Step 4 `npm run mongo:dev` console output:
```
> habitica@4.166.0 mongo:dev /Users/xxx/work/habitica
> run-rs -v 4.2.8 -l ubuntu1804 --keep --dbpath mongodb-data --number 1 --quiet

Skipping purge
Running '/Users/xxx/work/habitica/node_modules/run-rs/4.2.8/mongod' [ 27017 ]
Restarting replica set...
MongoError: This node was not started with the replSet option
    at Connection.<anonymous> (/Users/xxx/work/habitica/node_modules/mongodb-core/lib/connection/pool.js:443:61)
    at Connection.emit (events.js:314:20)
    at processMessage (/Users/xxx/work/habitica/node_modules/mongodb-core/lib/connection/connection.js:364:10)
    at Socket.<anonymous> (/Users/xxx/work/habitica/node_modules/mongodb-core/lib/connection/connection.js:533:15)
    at Socket.emit (events.js:314:20)
    at addChunk (_stream_readable.js:298:12)
    at readableAddChunk (_stream_readable.js:273:9)
    at Socket.Readable.push (_stream_readable.js:214:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)
```
Step 5 `npm run test:api-v3:integration` console output:
```
> habitica@4.166.0 test:api-v3:integration /Users/xxx/work/habitica
> gulp test:api-v3:integration

[14:38:11] Using gulpfile ~/work/habitica/gulpfile.js
[14:38:11] Starting 'test:api-v3:integration'...
[14:38:11] Starting 'test:prepare:mongo'...
[14:38:41] 'test:prepare:mongo' errored after 30 s
[14:38:41] Error: Unable to connect to mongo database. Are you sure it's running? 

MongooseServerSelectionError: Server selection timed out after 30000 ms
    at formatError (/Users/xxx/work/habitica/node_modules/gulp/node_modules/gulp-cli/lib/versioned/^4.0.0/format-error.js:21:10)
    at Gulp.<anonymous> (/Users/xxx/work/habitica/node_modules/gulp/node_modules/gulp-cli/lib/versioned/^4.0.0/log/events.js:33:15)
    at Gulp.emit (events.js:326:22)
    at Gulp.EventEmitter.emit (domain.js:483:12)
    at Object.error (/Users/xxx/work/habitica/node_modules/undertaker/lib/helpers/createExtensions.js:61:10)
    at handler (/Users/xxx/work/habitica/node_modules/now-and-later/lib/mapSeries.js:47:14)
    at f (/Users/xxx/work/habitica/node_modules/once/once.js:25:25)
    at f (/Users/xxx/work/habitica/node_modules/once/once.js:25:25)
    at tryCatch (/Users/xxx/work/habitica/node_modules/async-done/index.js:24:15)
    at cb (/Users/xxx/work/habitica/node_modules/async-done/index.js:40:12)
    at /Users/xxx/work/habitica/gulp/gulp-tests.js:63:21
    at /Users/xxx/work/habitica/node_modules/mongoose/lib/helpers/promiseOrCallback.js:16:11
    at /Users/xxx/work/habitica/node_modules/mongoose/lib/index.js:344:16
    at /Users/xxx/work/habitica/node_modules/mongoose/lib/connection.js:823:14
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
[14:38:41] 'test:api-v3:integration' errored after 30 s
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! habitica@4.166.0 test:api-v3:integration: `gulp test:api-v3:integration`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the habitica@4.166.0 test:api-v3:integration script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/xxx/.npm/_logs/2020-10-28T12_38_41_199Z-debug.log
```

**Comment by paglias** [comment]
@hamboomger we recently ( a few months back ) changed the way Mongo is setup, if you copy the NODE_DB_URI and TEST_DB_URI configurations from config.json.example and use them in your local config.json file it should work:

```
  "NODE_DB_URI": "mongodb://localhost:27017/habitica-dev?replicaSet=rs",
  "TEST_DB_URI": "mongodb://localhost:27017/habitica-test?replicaSet=rs",
```

**Comment by hamboomger** [comment]
@paglias maybe I should've pointed out that I already copied `config.json.example` to `config.json`. The lines you are describing are already in place. Does it mean that the test suit is working fine on your PC? :)

**Comment by paglias** [comment]
@hamboomger If you run `mongo` or `mongod` in your terminal, what's the output? Could you post the output of `npm ls run-rs`? The version needed is `0.6.2`, if you're on a different version install the correct one with `npm i run-rs@0.6.2`

The issue is not with the tests but with the database not starting correctly but I can't replicate it

**Comment by hamboomger** [comment]
@paglias as I said, I did not install the service on purpose, so the output is `bash: mongo: command not found` for both `mongo` and `mongod`. I checked in `package-lock.json` and I have version `0.6.2`:
![image](https://user-images.githubusercontent.com/9994683/97477658-8ee32780-1958-11eb-9331-1348e4fa4c33.png)

Maybe we have different OS versions and `run-rs` is simply not working on MacOS Mojave? I tried to run just `run-rs -l ubuntu1804` and still got the same error.

**Comment by paglias** [comment]
Hmm that's weird, I have the same config on mac (catalina) and it's working fine. I'd try to delete the `mongodb-data` folder and try again

**Comment by paglias** [comment]
Hi @hamboomger , did you try deleting mongodb-data?

**Comment by hamboomger** [comment]
@paglias ok, so I tried to run `run-rs` command today and it magically worked. `npm run mongo:dev` seems to be working as well and the `MongoError: This node was not started with the replSet option` is gone. I have no idea what's changed, becase I didn't merge any new changes from the develop branch this whole time, and `run-rs` library version is still the same =_=

So basically we can close this issue and I will continue investigation on my side.

**Comment by paglias** [comment]
Thanks for the heads up @hamboomger , unfortunately this type of bugs happen from time to time but they're hardly reproducible

**Comment by christinabranson** [comment]
I just wanted to confirm this issue & the (magic) solution on an identical setup as @hamboomger 

**Set Up**
- MacOS Mojave 10.14.6
- Fresh install of Habitica (forked today and followed the instructions in Setting up Habitica Locally on MacOS up to `npm run mongo:dev`)
- No prior installation of MongoDB

**Error**
Running `npm run mongo:dev` results in an error, namely `MongoError: This node was not started with the replSet option`.

**Solution**
After finding this thread, I only took 2 steps to try to resolve the issue:
1. Checked my version of run-rs. It had installed `0.7.4` and I manually downgraded to `0.6.2` with `npm i run-rs@0.6.2`, as provided above.
2. I restarted my machine.

Step 2 did the trick, and as hambooger reported, "it magically worked". `npm run mongo:dev` no longer produces an error and I am able to start the local instance within my browser.
